File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/domain/OSOutcomeEventsRepository.java
package com.onesignal.outcomes.domain;

import com.onesignal.OneSignalApiResponseHandler;
import com.onesignal.influence.model.OSInfluence;
import com.onesignal.outcomes.model.OSOutcomeEventParams;

import java.util.List;
import java.util.Set;

public interface OSOutcomeEventsRepository {

    List<OSOutcomeEventParams> getSavedOutcomeEvents();

    void saveOutcomeEvent(OSOutcomeEventParams event);

    void removeEvent(OSOutcomeEventParams outcomeEvent);

    void requestMeasureOutcomeEvent(String appId, int deviceType, OSOutcomeEventParams event, OneSignalApiResponseHandler responseHandler);

    void saveUniqueOutcomeNotifications(OSOutcomeEventParams eventParams);

    List<OSInfluence> getNotCachedUniqueOutcome(String name, List<OSInfluence> influences);

    Set<String> getUnattributedUniqueOutcomeEventsSent();

    void saveUnattributedUniqueOutcomeEventsSent(Set<String> unattributedUniqueOutcomeEvents);
}

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/domain/OSOutcomeEventsRepository.kt
package com.onesignal.outcomes.domain

import com.onesignal.OneSignalApiResponseHandler
import com.onesignal.influence.domain.OSInfluence

interface OSOutcomeEventsRepository {
    fun getSavedOutcomeEvents(): List<OSOutcomeEventParams>
    fun saveOutcomeEvent(event: OSOutcomeEventParams)
    fun removeEvent(outcomeEvent: OSOutcomeEventParams)
    fun requestMeasureOutcomeEvent(appId: String, deviceType: Int, event: OSOutcomeEventParams, responseHandler: OneSignalApiResponseHandler)
    fun saveUniqueOutcomeNotifications(eventParams: OSOutcomeEventParams)
    fun getNotCachedUniqueOutcome(name: String, influences: List<OSInfluence>): List<OSInfluence>
    fun getUnattributedUniqueOutcomeEventsSent(): Set<String>?
    fun saveUnattributedUniqueOutcomeEventsSent(unattributedUniqueOutcomeEvents: Set<String>)
    fun cleanCachedUniqueOutcomeEventNotifications(notificationTableName: String, notificationIdColumnName: String);
}